rotkehlchen/exchanges/kraken.py

```diff
@@ -781,3 +802,225 @@ def query_online_income_loss_expense(
+            base_trade, quote_trade, fee_part, kfee_part = None, None, None, None
+            for trade_part in trade_parts:
+                part_asset = trade_part.amount.asset
+                if trade_part.event_subtype == HistoryEventSubType.FEE:
+                    fee_part = trade_part
+                elif trade_part.amount.asset == A_KFEE:
+                    kfee_part = trade_part
+                elif part_asset == base_asset:
+                    base_trade = trade_part
+                elif part_asset == quote_asset:
+                    quote_trade = trade_part
+
+            if base_trade is None or quote_trade is None:
+                log.error(f'Failed to process {event_id} with parts {trade_parts}')
+                self.msg_aggregator.add_error(
+                    f'Failed to read trades for event {event_id}. '
+                    f'More details are available at the logs',
+                )
+                continue
+
+            base_amount = base_trade.amount.balance.amount
+            quote_amount = quote_trade.amount.balance.amount
+
+            if quote_amount < 0:
+                trade_type = TradeType.SELL
+                amount = base_amount
+                received_asset = quote_trade.amount.asset
+            else:
+                trade_type = TradeType.BUY
+                amount = quote_amount
+                received_asset = base_trade.amount.asset
+
+            if base_amount != ZERO:
+                rate = Price(quote_amount / base_amount)
+            else:
+                self.msg_aggregator.add_error(
+                    f'Rate for kraken trade couldnt be calculated. Base amount is ZERO '
+                    f'for event {event_id}',
+                )
+                continue
+
+            exchange_uuid = (
+                str(event_id) +
+                str(timestamp)
+            )
+            # If kfee was found we use it as the fee for the trade
+            if kfee_part is not None:
+                fee = Fee(kfee_part.amount.balance.amount)
+                fee_asset = A_KFEE
+            elif (None, None) == (fee_part, kfee_part):
+                fee = Fee(ZERO)
+                fee_asset = received_asset
+            elif fee_part is not None:
+                fee = Fee(fee_part.amount.balance.amount)
+                fee_asset = fee_part.amount.asset
+
+            trade = Trade(
+                timestamp=timestamp,
+                location=Location.KRAKEN,
+                base_asset=base_asset,
+                quote_asset=quote_asset,
+                trade_type=trade_type,
+                amount=AssetAmount(amount),
+                rate=rate,
+                fee=fee,
+                fee_currency=fee_asset,
+                link=exchange_uuid,
+            )
+            trades.append(trade)
+            max_ts = max(max_ts, timestamp)
+        return trades, Timestamp(max_ts)
+
+    @protect_with_lock()
+    def query_kraken_ledgers(self) -> bool:
```

Comment:
This is the concern I brought you yesterday. If we let this be retrieved by arguments we end up querying twice the same range. This function if you look at the definition uses its own query range to avoid querying from the beginning to now each time. The filter for time is applied later when reconstruction trades and deposits/withdrawals from the base entry events.